src/processor_x86.cpp

```diff
@@ -202,6 +203,7 @@ constexpr auto icelake = cannonlake | get_feature_masks(avx512bitalg, vaes, avx5
 constexpr auto icelake_server = icelake | get_feature_masks(pconfig, wbnoinvd);
 constexpr auto tigerlake = icelake | get_feature_masks(avx512vp2intersect, movdiri,
                                                        movdir64b, shstk);
+constexpr auto alderlake = skylake | get_feature_masks(avxvnni);
```

Comment:
Assuming this is correct, Intel really dialed back on the feature set in the client-class processors didn't they......

Reply:
On one way, yeah they reduced the feature set. But they went the AMD way of instead of adding more and more instructions, to make the ones that are already there faster.
The performance cores do have some AVX512 silicon in there but it isn't full avx512, and it's also fused off, apparently because the efficiency cores don't have them and that probably would be a mess to schedule.

Reply:
Actually, based on the post at https://github.com/JuliaLang/julia/issues/44691 this should also include

`clwb, sha, waitpkg, shstk, gfni, vaes, vpclmulqdq, rdpid, movdiri, movdir64b, serialize, ptwrite`

So it seems that nothing much was removed other than avx512.

If the list you gave is to be trusted, then it should also include

`pconfig`

Additionally, LLVM claims that this has `cldemote` and `pkg`. What's your `/proc/cpuinfo` (assuming linux of course....) I wonder if we are detection those incorrectly...